pybamm/solvers/base_solver.py

```diff
@@ -573,22 +589,22 @@ def solve(self, model, t_eval=None, external_variables=None, inputs=None):
                 # If the new initial conditions are different, set up again
                 # Doing the whole setup again might be slow, but no need to prematurely
                 # optimize this
-                self.set_up(model, ext_and_inputs, t_eval)
+                self.set_up(model, ext_and_inputs_list[0], t_eval)
                 self.models_set_up[model][
                     "initial conditions"
                 ] = model.concatenated_initial_conditions
         set_up_time = timer.time()
         timer.reset()
 
         # (Re-)calculate consistent initial conditions
-        self._set_initial_conditions(model, ext_and_inputs, update_rhs=True)
+        self._set_initial_conditions(model, ext_and_inputs_list[0], update_rhs=True)
 
         # Non-dimensionalise time
         t_eval_dimensionless = t_eval / model.timescale_eval
 
         # Calculate discontinuities
         discontinuities = [
-            event.expression.evaluate(inputs=inputs)
+            event.expression.evaluate(inputs=inputs_list[0])
```

Comment:
Generally, the whole preprocessing (setup + calculation of discontinuities) is made using the first element of the `inputs` list given as an argument to solve.

Maybe a comment explaining why only the first element of both `inputs_list` in `ext_and_inputs_list` could help